src/main.rs

```diff
@@ -1,12 +1,78 @@
-use std::io::Read;
+use jotdown::{html, Parser};
+use std::{
+    ffi::OsString,
+    fs::{self, File},
+    io::{self, BufWriter},
+    process::exit,
+};
+
+#[derive(Default)]
+struct App {
+    input: Option<OsString>,
+    output: Option<OsString>,
+}
+
+fn parse_args() -> App {
+    let mut app = App::default();
+
+    let mut args = std::env::args_os().skip(1).peekable();
+
+    while let Some(arg) = args.next() {
+        match (arg.to_string_lossy().as_ref(), args.peek()) {
+            ("-h" | "--help", _) => {
+                eprint!("{}", include_str!("./help.txt"));
+                exit(0);
+            }
+            ("-v" | "--version", _) => {
+                eprintln!("jotdown v{}", env!("CARGO_PKG_VERSION"));
+                exit(0);
+            }
+            (flag @ "-o" | flag @ "--output", o) => match o {
+                Some(o) => {
+                    app.output = Some(o.into());
+                    args.next();
+                }
+                None => {
+                    eprintln!("Please supply an argument to {flag}");
+                    exit(1);
+                }
+            },
+            (file, _) if !file.starts_with('-') => app.input = Some(file.into()),
+            ("-", _) => {}
+            (flag, _) => {
+                eprintln!("Unknown flag: {flag}");
+                exit(1)
```

Comment:
could print the usage here also, as one may have incorrectly guessed a flag

Reply:
Not a bad idea, I'll add that!